TST: quantity ufunc has 4 failures in numpy 2.0.0.dev (but really it  is 1.26.dev)

Numpy: 2.0.0.dev0+279.gdbc454a15

astropy/units/tests/test_quantity_non_ufuncs.py

```
__________________________ test_testing_completeness ___________________________

    @needs_array_function
    def test_testing_completeness():
        assert not CoverageMeta.covered.intersection(untested_functions)
>       assert all_wrapped == (CoverageMeta.covered | untested_functions)
E       assert {<function ar...a758130>, ...} == {<function ar...6d6e130>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function merge_arrays at 0x7f35746891f0>,
E            <function drop_fields at 0x7f35746898b0>,
...
```

```
______________ TestFunctionHelpersCompleteness.test_all_included _______________

self = <astropy.units.tests.test_quantity_non_ufuncs.TestFunctionHelpersCompleteness object at ...>

    @needs_array_function
    def test_all_included(self):
        included_in_helpers = (
            SUBCLASS_SAFE_FUNCTIONS
            | UNSUPPORTED_FUNCTIONS
            | set(FUNCTION_HELPERS.keys())
            | set(DISPATCHED_FUNCTIONS.keys())
        )
>       assert all_wrapped == included_in_helpers
E       assert {<function ar...a758130>, ...} == {<function ar...6d48170>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function merge_arrays at 0x7f35746891f0>,
E            <function drop_fields at 0x7f35746898b0>,
...
```

astropy/utils/masked/tests/test_function_helpers.py

```
_______________________ test_basic_testing_completeness ________________________

    def test_basic_testing_completeness():
>       assert all_wrapped == (tested_functions | IGNORED_FUNCTIONS | UNSUPPORTED_FUNCTIONS)
E       assert {<function ar...6da0130>, ...} == {<function ar...a758130>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function broadcast_to at 0x7f3576d145b0>,
E            <function broadcast_arrays at 0x7f3576d147f0>,
...
```

```
______________ TestFunctionHelpersCompleteness.test_all_included _______________

self = <astropy.utils.masked.tests.test_function_helpers.TestFunctionHelpersCompleteness object at ...>

    def test_all_included(self):
        included_in_helpers = (
            MASKED_SAFE_FUNCTIONS
            | UNSUPPORTED_FUNCTIONS
            | set(APPLY_TO_BOTH_FUNCTIONS.keys())
            | set(DISPATCHED_FUNCTIONS.keys())
        )
>       assert all_wrapped == included_in_helpers
E       assert {<function ar...6da0130>, ...} == {<function ar...6d6e130>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function broadcast_to at 0x7f3576d145b0>,
E            <function broadcast_arrays at 0x7f3576d147f0>,
...
```

--- a/astropy/units/tests/test_quantity_non_ufuncs.py
+++ b/astropy/units/tests/test_quantity_non_ufuncs.py
@@ -17,7 +17,7 @@ from astropy.units.quantity_helper.function_helpers import (
     TBD_FUNCTIONS,
     UNSUPPORTED_FUNCTIONS,
 )
-from astropy.utils.compat import NUMPY_LT_1_23, NUMPY_LT_1_24
+from astropy.utils.compat import NUMPY_LT_1_23, NUMPY_LT_1_24, NUMPY_LT_1_25
 
 needs_array_function = pytest.mark.xfail(
     not ARRAY_FUNCTION_ENABLED, reason="Needs __array_function__ support"
@@ -25,16 +25,26 @@ needs_array_function = pytest.mark.xfail(
 
 
 # To get the functions that could be covered, we look for those that
-# are wrapped.  Of course, this does not give a full list pre-1.17.
+# are in modules we care about and have been overridden.
 def get_wrapped_functions(*modules):
-    wrapped_functions = {}
-    for mod in modules:
-        for name, f in mod.__dict__.items():
-            if f is np.printoptions or name.startswith("_"):
-                continue
-            if callable(f) and hasattr(f, "__wrapped__"):
-                wrapped_functions[name] = f
-    return wrapped_functions
+    if NUMPY_LT_1_25:
+
+        def allows_array_function_override(f):
+            return (
+                hasattr(f, "__wrapped__")
+                and f is not np.printoptions
+                and not f.__name__.startswith("_")
+            )
+
+    else:
+        from numpy.testing.overrides import allows_array_function_override
+
+    return {
+        name: f
+        for mod in modules
+        for name, f in mod.__dict__.items()
+        if callable(f) and allows_array_function_override(f)
+    }
 
 
 all_wrapped_functions = get_wrapped_functions(

